moveit_planners/pilz_industrial_motion_planner/include/joint_limits/joint_limits_rosparam.hpp

```diff
@@ -27,6 +27,16 @@
 #include "joint_limits/joint_limits.hpp"
 #include "rclcpp/rclcpp.hpp"
 
+/** declare a parameter if not already declared. */
+#define DECLARE(node, name, type, default_value)                                                                       \
+  do                                                                                                                   \
+  {                                                                                                                    \
+    if (not node->has_parameter(name))                                                                                 \
+    {                                                                                                                  \
+      node->declare_parameter<type>(name, default_value);                                                              \
+    }                                                                                                                  \
+  } while (false);
+
```

Comment:
I don't understand the need for the do-while?

Reply:
I didn't understand this either.

Reply:
By habit. Should I remove it?

Reply:
Yeah I would prefer just a clean if statement here.

Reply:
I would generally really prefer a template function over a c-macro.  It should be fairly straightforward to make this a template function.